### PR TITLE
[MM-44422] Open Members RHS from Channel dropdown

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
This PR allows users to open the Members RHS from the channel dropdown. In the dropdown there are two different buttons:

  - Manage Members: This will open the member RHS in `edit` mode. To achieve that I had to move the editing state from the component to the redux store
  - View Members: Just open the RHS.

Going through the e2e tests I realized those 2 things that I fixed:
  1. Guests must have a badge by their name
  2. You cannot manage an archived channel.

I did not remove the modal **yet** as I might need for some future improvement on this component

I created a guest user in the spinwick so you can see it in the member RHS with the guest badge

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44422

#### Related Pull Requests
N/A

#### Screenshots
N/A 

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: reducers/views/rhs.ts [hunk 1]
```diff
@@ -350,6 +350,23 @@ function isMenuOpen(state = false, action: GenericAction) {
...
+    case ActionTypes.SET_EDIT_CHANNEL_MEMBERS:
+        return action.active;
+    case ActionTypes.UPDATE_RHS_STATE:
+        if (!action.state) {
+            return false;
+        }
+        return state;
+
+    case UserTypes.LOGOUT_SUCCESS:
+        return false;
```
Comment (marianunez) [inline reducers/views/rhs.ts:364]: Is this intended here?

File: components/channel_header_dropdown/menu_items/toggle_members_rhs/toggle_members_rhs.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    textOpen: string;
+    textClose: string;
+}
+
+type Props = {
+    rhsOpen: boolean;
+    actions: Action;
+} & OwnProps;
+
+const ToggleChannelMembersRHS = ({
```
Comment (marianunez) [inline components/channel_header_dropdown/menu_items/toggle_members_rhs/toggle_members_rhs.tsx:28]: Should we add a cypress test to cover this behavior?

**Review by marianunez — COMMENTED** [review]
Looking great! Mostly some questions below about the expected behavior

File: reducers/views/rhs.ts (hunk 1 above)
Comment (JulienTant) [reply reducers/views/rhs.ts:364]: It seems that all reducers here took care of this case so I figured why not 😅

File: components/channel_header_dropdown/channel_header_dropdown_items.js [hunk 3]
```diff
@@ -164,13 +164,12 @@ export default class ChannelHeaderDropdown extends React.PureComponent {
...
+                    <MenuItemToggleMembersRHS
                         id='channelViewMembers'
+                        channel={channel}
                         show={channel.type !== Constants.DM_CHANNEL && channel.type !== Constants.GM_CHANNEL && (isArchived || isDefault)}
-                        modalId={ModalIdentifiers.CHANNEL_MEMBERS}
-                        dialogType={ChannelMembersModal}
-                        dialogProps={{channel}}
-                        text={localizeMessage('channel_header.viewMembers', 'View Members')}
+                        textOpen={localizeMessage('channel_header.viewMembers', 'View Members')}
+                        textClose={localizeMessage('channelHeader.closeMembersRHS', 'Close Members RHS')}
```
[the remark this answers is not in the document]
Comment (JulienTant) [reply components/channel_header_dropdown/channel_header_dropdown_items.js:172]: Happy to change it to anything better, I don't remember seeing a design associated to this. FYI the "Show info" at the top turns into "Hide info", at first I've set this to "Hide members" but it felt weird.

File: components/channel_header_dropdown/menu_items/toggle_members_rhs/toggle_members_rhs.tsx (hunk 2 above)
Comment (JulienTant) [reply components/channel_header_dropdown/menu_items/toggle_members_rhs/toggle_members_rhs.tsx:28]: Yes that's a good catch, I've changed so much e2e tests covering this menu option (both View and Manage members) that I didn't think about creating dedicated tests

**Review by julmondragon — APPROVED** [review]
Looks good!

File: components/channel_header_dropdown/menu_items/toggle_members_rhs/toggle_members_rhs.tsx (hunk 2 above)
Comment (JulienTant) [reply components/channel_header_dropdown/menu_items/toggle_members_rhs/toggle_members_rhs.tsx:28]: change applied: I added a test case for both 'View Members' and 'Manage Members', and also added a missing case for clicking on the 'Members icon' in the channel header. Thanks for suggesting those changes!

**Review by stevemudie — APPROVED** [review]
Looks good @JulienTant 
Both work as expected.

File: components/channel_header_dropdown/channel_header_dropdown_items.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (esethna) [reply components/channel_header_dropdown/channel_header_dropdown_items.js:172]: I would propose we keep the text static and if the RHS is already open to the members pane, then do nothing.

It seems pretty odd to me to use a menu item to close the RHS pane, but we might not want to remove it from the menu to accommodate muscle memory for that item
.

**Pull request merged**